cty/json/value.go

```diff
@@ -61,21 +61,5 @@ func Marshal(val cty.Value, t cty.Type) ([]byte, error) {
 // may be a cty.PathError.
 func Unmarshal(buf []byte, t cty.Type) (cty.Value, error) {
 	var path cty.Path
-	return unmarshal(buf, t, path, &unmarshalOptions{})
-}
-
-// UnmarshalDynamicWithImpliedType decodes a JSON representation of the
-// given value into a cty.Value conforming to the given type while replacing
-// dynamicPseudoType attributes with their implied type.
-//
-// Implied types are a best-effort deduction of the type from the structure
-// of the input JSON.
-//
-// While decoding, type conversions will be done where possible to make
-// the result conformant even if the types given in JSON are not exactly
-// correct. If conversion isn't possible then an error is returned, which
-// may be a cty.PathError.
```

Comment:
I always assumed that go-cty was responsible for implying the result for Terraform arguments, but it must be something built into HCL 😄. Interesting!